lib/improver/tests/blending/calculate_weights_and_blend/test_WeightAndBlend.py

```diff
@@ -0,0 +1,425 @@
+    "uk_ens": {"forecast_period": [0, 4, 8],
+               "weights": [0, 1, 1],
+               "units": "hours"}
+}
+
+
+def set_up_masked_cubes():
+    """
+    Set up cubes with masked data for spatial weights tests
+
+    Returns:
+        iris.cube.CubeList:
+            List containing a UKV cube with some rain, and a masked nowcast
+            cube with more rain.
+    """
+    thresholds = np.array([0.5, 1, 2], dtype=np.float32)
+    units = "mm h-1"
+    name = "lwe_precipitation_rate"
+    datatime = dt(2018, 9, 10, 7)
+    cycletime = dt(2018, 9, 10, 5)
+
+    # 5x5 matrix results in grid spacing of 200 km
+    base_data = np.ones((5, 5), dtype=np.float32)
+
+    # set up a UKV cube with some rain
+    rain_data = np.array([0.9*base_data, 0.5*base_data, 0*base_data])
+    ukv_cube = set_up_probability_cube(
+        rain_data, thresholds, variable_name=name,
+        threshold_units=units, time=datatime, frt=cycletime,
+        spatial_grid="equalarea", standard_grid_metadata="uk_det")
+
+    # set up a masked nowcast cube with more rain
+    more_rain_data = np.array([base_data, 0.6*base_data, 0.2*base_data])
+    radar_mask = np.broadcast_to(
+        np.array([False, False, False, True, True]), (3, 5, 5))
+    more_rain_data = np.ma.MaskedArray(more_rain_data, mask=radar_mask)
+    nowcast_cube = set_up_probability_cube(
+        more_rain_data, thresholds, variable_name=name,
+        threshold_units=units, time=datatime, frt=cycletime,
+        spatial_grid="equalarea",
+        attributes={"mosg__model_configuration": "nc_det"})
+
+    return iris.cube.CubeList([ukv_cube, nowcast_cube])
+
+
+class Test__init__(IrisTest):
+    """Test the __init__ method"""
+
+    def test_cycle(self):
+        """Test initialisation for cycle blending"""
+        plugin = WeightAndBlend(
+            "forecast_reference_time", "linear", y0val=1, ynval=1)
+        self.assertEqual(plugin.blend_coord, "forecast_reference_time")
+        self.assertEqual(plugin.wts_calc_method, "linear")
+        self.assertIsNone(plugin.weighting_coord)
+        self.assertAlmostEqual(plugin.y0val, 1.0)
+        self.assertAlmostEqual(plugin.ynval, 1.0)
+
+    def test_model(self):
+        """Test initialisation for model blending"""
+        plugin = WeightAndBlend(
+            "model_id", "dict", weighting_coord="forecast_period",
+            wts_dict=MODEL_WEIGHTS)
+        self.assertEqual(plugin.blend_coord, "model_id")
+        self.assertEqual(plugin.wts_calc_method, "dict")
+        self.assertEqual(plugin.weighting_coord, "forecast_period")
+        self.assertDictEqual(plugin.wts_dict, MODEL_WEIGHTS)
+
+    def test_unrecognised_weighting_type_error(self):
+        """Test error if wts_calc_method is unrecognised"""
+        msg = "Weights calculation method 'kludge' unrecognised"
+        with self.assertRaisesRegex(ValueError, msg):
+            WeightAndBlend("forecast_period", "kludge")
+
+
+class Test__calculate_blending_weights(IrisTest):
```

Comment:
I think we need at least one test that makes use of the sorting functionality. At least with the dict option, preferably with another method, e.g. linear as well.

Reply:
I'm not sure I can make a meaningful test for the dictionary option (only used for model blending), as this will never actually trigger sorting.  The process is to call `MergeCubesForWeightedBlending`, which constructs the numerical `model_id` coordinate in ascending order, so the sorting has no effect.  But I can make a test for cycle blending.

Reply:
In fact I'm not sure this sort statement is ever triggered, given the way `MergeCubesForWeightedBlending` behaves (ie producing a monotonic blend coordinate).  I'll remove it and see what happens...

Reply:
I've confirmed the sorting isn't used by adding a test to `MergeCubesForWeightedBlending`.  All tests continue to pass when the sorting is removed.